locale/en/docs/guides/backpressuring-in-streams.md

```diff
@@ -0,0 +1,480 @@
+
+## Lifecycle of `.pipe()`
+
+To achieve a better understanding of backpressure, here is a flow-chart on the
+lifecycle of a [`Readable`][] stream being [piped][] into a [`Writable`][]
+stream:
+
+```javascript
++===============+               +===============+
+|   Your Data   |               |  .pipe(dest)  |
++===============+               |===============|
+        |                       |  Setup event  |
+        |                       |  closures:    |
++-------v-----------+           |               |         +-------------------+
+|  Readable Stream  >-----------+ * .data(cb)   +--------->  Writable Stream  |
++-^-------^-------^-+           | * .drain(cb)  |         +-------------------+
+  ^       ^       ^             | * .unpipe(cb) |                   |
+  |       ^       |             | * .error(cb)  |                   |
+  |       |       |             | * .finish(cb) |                   |
+  ^       |       |             | * .end(cb)    |           +=======v=========+
+  ^       |       ^             | * .close(cb)  |           |  .write(chunk)  |
+  |       |       ^             +---------------+           +=======+=========+
+  |       ^       |                                                 |
+  |       ^       |                              +------------------v---------+
+  ^       |       +-> if (!chunk)                |    Is this chunk too big?  |
+  ^       |       |     emit .end();             |    Is the queue busy?      |
+  |       |       +-> else                       +-------+----------------+---+
+  |       ^       |     emit .write();                   |                |
+  |       ^       ^                                   +--v---+        +---v---+
+  |       |       ^-----------------------------------<  No  |        |  Yes  |
+  ^       |                                           +------+        +---v---+
+  ^       |                                                               |
+  |       ^               emit .pause();          +=================+     |
+  |       ^---------------^-----------------------+  return false;  <-----+---+
+  |                                               +=================+         |
+  |                                                                           |
+  ^            when queue is empty     +============+                         |
+  ^------------^-----------------------<  Buffering |                         |
+               |                       |============|                         |
+               +> emit .drain();       |  ^Buffer^  |                         |
+               +> emit .resume();      +------------+                         |
+                                       |  ^Buffer^  |                         |
+                                       +------------+   add chunk to queue    |
+                                       |            <---^---------------------<
+                                       +============+
+```
```

Comment:
Wondering if by this you meant something like:

```
                                                     +===================+
                         +-->  Piping functions  +--->   src.pipe(dest)  |
                         x     are set up during     |===================|
                         x     the .pipe method.     |  Event callbacks  |
+===============+        x                           |-------------------|
|   Your Data   |        x     They exist outside    | .on('close', cb)  |
+=======+=======+        x     the data flow, but    | .on('data', cb)   |
        |                x     importantly attach    | .on('drain', cb)  |
        |                x     events, and their     | .on('unpipe', cb) |
+-------v-----------+    x     respective callbacks. | .on('error', cb)  |
|  Readable Stream  +----+                           | .on('finish', cb) |
+-^-------^-------^-+    |                           | .on('end', cb)    |
  ^       |       ^      |                           +-------------------+
  |       |       |      |                           
  |       ^       |      |                                              
  ^       ^       ^      |    +-------------------+         +=================+
  ^       |       ^      +---->  Writable Stream  +--------->  .write(chunk)  |
  |       |       |           +-------------------+         +=======+=========+
```

Or even have more explicit steps come in?